### PR TITLE
feat: cloneNode, importNode, adoptNode

### PR DESCRIPTION
**Pull request by morille**
https://developer.mozilla.org/en-US/docs/Web/API/Node/cloneNode
https://developer.mozilla.org/en-US/docs/Web/API/Document/importNode
https://developer.mozilla.org/en-US/docs/Web/API/Document/adoptNode

### CODE REVIEW

File: src/dom/node.ts [hunk 1]
```diff
@@ -137,12 +137,14 @@ export class Node extends EventTarget {
...
+  _setOwnerDocument(document: Document | null, deep: boolean = true) {
     if (this.#ownerDocument !== document) {
       this.#ownerDocument = document;
 
-      for (const child of this.childNodes) {
-        child._setOwnerDocument(document);
+      if (deep) {
+        for (const child of this.childNodes) {
+          child._setOwnerDocument(document);
+        }
```
[the remark this answers is not in the document]
Comment (morille) [reply src/dom/node.ts:147]: You're absolutely right, reverted that.

File: test/units/cloneNode.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { DOMParser } from "../../deno-dom-wasm.ts";
+import { Element } from "../../src/dom/element.ts";
+import { Node } from "../../src/dom/node.ts";
```
Comment (b-fuze) [inline test/units/cloneNode.ts:3]: Sorry, I missed this, but please import both `Element` and `Node` from `deno-dom-wasm.ts` and not from anything in `src/`

File: test/units/cloneNode.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  checkClone(doc, doc.cloneNode(true));
+});
+
+function checkClone(node: Node, clone: Node) {
+  assert(clone instanceof Node);
+  assert(node !== clone);
+  if (node instanceof Element) {
+    assertEquals((node as Element).outerHTML, (clone as Element).outerHTML);
+  }
+  else {
```
Comment (b-fuze) [inline test/units/cloneNode.ts:23]: The `else` and closing brace should be on the same line

**Comment by morille** [comment]
@b-fuze Thanks. BTW do you have the option on github to squash everything before merge or I should do it on my side ?

**Comment by b-fuze** [comment]
I can squash it, yeah

**Pull request merged**